js/src/annotations/osd-canvas-renderer.js

```diff
@@ -357,6 +371,19 @@
         display.remove(); //remove this annotation display from dom
       });
 
```

Comment:
This is where the magic happens. Toggle the status of the pin and of the actual annotation, so that it's no longer closed on `mouseleave`.